src/util/cout_message.cpp

```diff
@@ -88,12 +88,7 @@ void console_message_handlert::flush(unsigned level)
   // We flush after messages of level 6 or lower.
   // We don't for messages of level 7 or higher to improve performance,
   // in particular when writing to NFS.
-  if(level>=4)
-  {
-    if(level<=6)
-      std::cout << std::flush;
-  }
-  else
+  if(always_flush || level <= 6)
```

Comment:
That does not seem to be correct: the previous code would (intentionally) flush `std::cout` in some levels as that's where output is sent to.